### Задача 1.  
Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
Найдите ее среднее значение и дисперсию.  
$Решение:$  
Для определения мат. ожидания и дисперсии случайной непрерывной величины , на отрезке [a,b], формулы:  
${Матем. ожидание} : M(A)= \frac{a+b}{2}\\{Дисперсия: }D(A)= \frac{(b-a)^2}{12}$

In [17]:
def math_exp_and_variance(a,b):
    return f'На промежутке ({a};{b}]\nСреднее значение: М(А) = {(a+b)/2: .2f}\nДисперсия: D(A) = {((b-a)**2)/12: .2f}'
print(math_exp_and_variance(200,800))

На промежутке (200;800]
Среднее значение: М(А) =  500.00
Дисперсия: D(A) =  30000.00


$ Ответ :$   
$ M(A) = 500.00$  
$ D(A) = 30000.00$


### Задача 2.  
О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?
Если да, найдите ее.  
$Решение:$  
Величину В находящуюся на отрезке (a,b), дисперсия случайной непрерывной равномерно распределенной находим по формуле:  
${Дисперсия}: D(A)= \frac{(b-a)^2}{12}$ Значения по условию: $Дисперсия$ $D(B-a)=0.2$ и левая граница отрезка $a=0.5$  
 Правая граница $b$ величины $B$ по формуле: $ D(B-b)\frac{(b-0.5)^2}{12}=0.2 b-0.5=\sqrt{2.4} b=0.5+\sqrt{2.4}$,  
 Среднее значение по формуле: $ M(B)=\frac{a+b}{2}$

In [18]:
b=0.5+2.4**(1/2)
print(f'Правая граница распределения величины В-b = {b: .3f}\n'
      f'Среднее значение В на промежутке (0.5; {b: .3f}) M(B) = {(b+0.5)/2: .3f}')

Правая граница распределения величины В-b =  2.049
Среднее значение В на промежутке (0.5;  2.049) M(B) =  1.275


$Ответ:$
Правая граница распределения: $b=2.049$
Среднее значение: $M(B)=1.275$

### Задача 3.  
Непрерывная случайная величина X распределена нормально и задана плотностью распределения: $f(x)=\frac{1}{4\cdot\sqrt{2\pi}}\cdot e^{-\frac{(x+2)^2}{32}}$    
$Найдите:$  
а). M(X)  
б). D(X)  
в). $\displaystyle\sigma$(X) (среднее квадратичное отклонение)  
Формула плотности нормального распределения:  
$f(x)=\frac{1}{\sigma\sqrt{2\pi}}\cdot e^{-\frac{(x-a)^2}{2\sigma^2}}$  
$M(X)=a=-2\sigma = 4D(X)=\sigma^2=16$  
      
$Ответ:$  
$а). M(X)=-2;$  
$б). D(X)=16;$  
$в). \sigma=4$

### Задача 4.
Рост взрослого населения города X имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
1. больше 182 см?
2. больше 190 см?
3. от 166 см до 190 см?
4. от 166 см до 182 см?
5. от 158 см до 190 см?
6. не выше 150 см или не ниже 190 см?
7. не выше 150 см или не ниже 198 см?
8. ниже 166 см?

Задачу можно решить двумя способами: без использования сторонних библиотек (numpy, scipy, pandas и пр.), а затем проверить себя с помощью встроенных функций.  
$Решение:$
Для значений Z нормального распределения, функция:

In [1]:
def z_value(height):
    return (height-174)/8

Расчет значений вероятности методом cdf, стандартный модуль statistics, объект NormalDist.

In [3]:
from statistics import NormalDist

#### 1) Определим значениe Z=?:

In [6]:
print(z_value(182))

1.0


Случайная величина X будет меньше или равна x, т.е в данном случае $P(X\le182)$ 

In [7]:
NormalDist().cdf(z_value(182))

0.8413447460685428

По условию нам нужна вероятность $P(X\ge182)$, поэтому вычтем полученное значение из $1$:

In [32]:
P=1-NormalDist().cdf(z_value(182))
P

0.15865525393145719

$P(X\ge182)=0.1587$

#### 2) Определим значениe Z=?:

In [33]:
z=z_value(190)
z

2.0

Определим $P(X\ge190)$

In [34]:
P=1-NormalDist().cdf(z)
P

0.022750131948179098

$P(X\ge190)$=0.02275

#### 3) Определим $Z_1$ и $Z_2$

In [35]:
z1=z_value(166)
z1

-1.0

In [36]:
z2=z_value(190)
z2

2.0

$P(166\le X\le 190)=0.8186$

#### 4) Определим $Z_1$ и $Z_2$

In [37]:
z1=z_value(166)
z1

-1.0

In [38]:
z2=z_value(182)
z2

1.0

$P(166\le X\le 182)$

In [39]:
P=NormalDist().cdf(z2)-NormalDist().cdf(z1)
P

0.6826894921370856

$P(166\le X\le 182)=0.6827$

#### 5) Определим $Z_1$ и $Z_2$

In [40]:
z1=z_value(158)
z1

-2.0

In [41]:
z2=z_value(190)
z2

2.0

Определим $P(158\le X\le 190)$

In [42]:
P=NormalDist().cdf(z2)-NormalDist().cdf(z1)
P

0.9544997361036418

$P(158\le X\le 190)=0.9545$

#### 6) Определим $Z_1$ и $Z_2$


In [45]:
z1=z_value(150)
z1

-3.0

In [46]:
z2=z_value(190)
z2

2.0

Определим $P(X\le150\cup X\ge 190)$

In [47]:
P=NormalDist().cdf(z1)+(1-NormalDist().cdf(z2))
P

0.0241000299798092

$P(X\le150\cup X\ge 190)=0.0241$

#### 7) Определим $Z_1$ и $Z_2$

In [48]:
z1=z_value(150)
z1

-3.0

In [49]:
z2=z_value(198)
z2

3.0

Определим $P(X\le150\cup X\ge 198)$

In [50]:
P=NormalDist().cdf(z1)+(1-NormalDist().cdf(z2))
P

0.002699796063260207

$P(X\le150\cup X\ge 198)=0.0027$

#### 8) Определим значениe Z=?:

In [52]:
z=z_value(166)
z

-1.0

Определим $P(X\le166)$

In [53]:
P=NormalDist().cdf(z)
P

0.15865525393145713

$P(X\le166)=0.1587$

#### Ответ:  
$а) P(X\ge182)=0.1587;$  
$б) P(X\ge190)=0.02275;$  
$в) P(166\le X\le 190)=0.8186;$  
$г) P(166\le X\le 182)=0.6827;$  
$д) P(158\le X\le 190)=0.9545;$  
$е) P(X\le150\cup X\ge 190)=0.0241;$  
$ё) P(X\le150\cup X\ge 198)=0.0027;$  
$ж) P(X\le166)=0.1587.$

### Задача 5.
На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

Воспользуемся формулой расчета $Z$-оценки:  
$\displaystyle Z=\frac{X-\bar x}{\sigma}=\frac{190-178}{\sqrt{25}}=\frac{12}{5}$

In [54]:
Z=(190-178)/25**(1/2)
Z

2.4

In [55]:
print(f'Рост человека, равный 190 см, отклоняется от математического ожидания роста в популяции, \n'
      f'в которой M(X) = 178 см и D(X) = 25 кв.см на Z = {Z} сигм.')

Рост человека, равный 190 см, отклоняется от математического ожидания роста в популяции, 
в которой M(X) = 178 см и D(X) = 25 кв.см на Z = 2.4 сигм.


#### Ответ:  
На $2.4$ сигмы, отклоняется рост человека, равный 190 см.